core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -732,17 +732,12 @@ class DynamicLogConfig(logManager: LogManager, server: KafkaBroker) extends Brok
   override def reconfigure(oldConfig: KafkaConfig, newConfig: KafkaConfig): Unit = {
     val originalLogConfig = logManager.currentDefaultConfig
     val originalUncleanLeaderElectionEnable = originalLogConfig.uncleanLeaderElectionEnable
-    val newBrokerDefaults = new util.HashMap[String, Object](originalLogConfig.originals)
-    newConfig.valuesFromThisConfig.forEach { (k, v) =>
-      if (DynamicLogConfig.ReconfigurableConfigs.contains(k)) {
-        DynamicLogConfig.KafkaConfigToLogConfigName.get(k).foreach { configName =>
-          if (v == null)
-             newBrokerDefaults.remove(configName)
-          else
-            newBrokerDefaults.put(configName, v.asInstanceOf[AnyRef])
-        }
+    val newBrokerDefaults = new util.HashMap[String, Object](newConfig.extractLogConfigMap)
+    originalLogConfig.originals().forEach((k, v) => {
+      if (!DynamicLogConfig.ReconfigurableConfigs.contains(k)) {
+        newBrokerDefaults.put(k, v)
```

Comment:
1. Do we need to translate the name from KafkaConfig to LogConfig as the original code does?
2. We started `newBrokerDefaults` with `newConfig.extractLogConfigMap` that includes non-reconfigurable configs. Where is the logic to remove them?

Reply:
Let me fix this a bit. It should be sufficient to just remove the non-reconfigurable configs (really, singular config)